public/app/core/utils/explore.ts

```diff
@@ -521,3 +521,35 @@ export const getQueryResponse = (
 ) => {
   return from(datasourceInstance.query(options, observer));
 };
+
+export const getShiftedTimeRange = (direction: number, origRange: TimeRange, timeZone: TimeZone): RawTimeRange => {
```

Comment:
This could probably be shared between Explore and Dashboards? Do we add this to grafana/data package?

Reply:
yes (on being shared), but don't think we need this in grafana/data yet, we could add this to core/utils for now. maybe a similar function to handle zoom, that also looks to be duplicated in DashNavControls and ExploreTimeControls

Reply:
Perfect thanks, I'll fix that!